etl.py

```diff
@@ -0,0 +1,171 @@
+import csv
+from datetime import datetime
+import json
+from typing import List, Any, Optional
+
+
+class ETLService:
+
+    def __init__(self):
+        self.columns_mapping = None
+        self.input_file_headers = None
+        self.output_file_headers = None
+
+    def run(self, input_file: str, output_file: str, columns_mapping_file: str) -> None:
+        """Performs the complete ETL process."""
+
+        input_data = self.read_csv(input_file)
+
+        input_data = self.extract_headers(input_data)
+
+        with open(columns_mapping_file, 'r') as f:
+            self.columns_mapping = json.load(f)
+
+        transformed_data = self.transform_data(input_data)
+
+        self.output_file_headers = [column for column in self.columns_mapping.values() if column is not None]
+
+        self.write_csv(output_file, transformed_data)
+
+    def extract_headers(self, input_data: List[List[Any]]):
+        """Save the headers before removing the first row and remove headers from data."""
+        self.input_file_headers = input_data[0] if input_data else []
+        if input_data:
+            input_data_without_headers = input_data[1:]
+            return input_data_without_headers
+        else:
+            return input_data
+
+    def read_csv(self, filename: str) -> List[List[Any]]:
+        """Reads a CSV file and returns the data as a list of rows."""
+        data: List[List[Any]] = []
+        try:
+            with open(filename, 'r', newline='') as file:
+                reader = csv.reader(file)
+                for row in reader:
+                    data.append(row)
```

Comment:
This could be a generator and just `yield` row to avoid building out a potentially large in-memory data structure.